tests/test_numbered_collection.py

```diff
@@ -115,6 +115,18 @@ def test_append_renumber(self):
         self.assertEqual(cell.number, 4)
         self.assertEqual(len(cells), size + 2)
 
+    def test_append_renumber_problems(self):
+        prob1 = copy.deepcopy(self.simple_problem)
+        prob2 = copy.deepcopy(self.simple_problem)
+        # Delete Material 2, making its number available.
+        prob2.materials.remove(prob2.materials[2])
+        len_mats = len(prob2.materials)
+        mat1 = prob1.materials[1]
+        new_num = prob2.materials.append_renumber(mat1)
+        self.assertEqual(new_num, 2)
+        self.assertEqual(len(prob2.materials), len_mats + 1)
+        self.assertIs(prob2.materials[2], mat1)
```

Comment:
We're actually transitioning from unittest to pytest, so new tests should be more pytest-y.

Switch to:

``` python
assert new_num == 2
assert len(prob2.materials) == len_mats + 1
assert prob2.materials[2] is mat1
```

Reply:
pytest is the pybest